addon/ng2/blueprints/component/files/__path__/__name__.component.spec.ts

```diff
@@ -10,14 +10,28 @@ import {
   ComponentFixture,
   TestComponentBuilder
 } from 'angular2/testing';
-import {provide} from 'angular2/core';
+import {provide} from 'angular2/core';<% if (route) { %>
+import {Router, RouteParams} from 'angular2/router';<% } %>
 import {<%= classifiedModuleName %>Component} from './<%= dasherizedModuleName %>.component';
+<% if (route) { %>
+class MockRouter {
+  navigate() { }
+  registerPrimaryOutlet() { }
+}
 
-describe('<%= classifiedModuleName %> Component', () => {
-
-  beforeEachProviders((): any[] => []);
-
+class MockRouteParams {
+  get() { return 1; }
+}<% } %>
 
+describe('<%= classifiedModuleName %> Component', () => {
+<% if (route) { %>
+  beforeEachProviders(() => [
+    provide(Router, { useClass: MockRouter }),
+    provide(RouteParams, { useClass: MockRouteParams }),
+  ]);
+<% } else { %>
+  beforeEachProviders((): any[] => [<% if (route) { %>ROUTER_PROVIDERS<% } %>]);
```

Comment:
This means you can probably do without `ROUTER_PROVIDERS` altogether and just get by mocking `MockRouter.registerPrimaryOutlet()`.